openhgnn/models/MAGNN.py

```diff
@@ -119,24 +123,58 @@ def __init__(self, in_feats, h_feats, inter_attn_feats, num_heads, num_classes,
         self.feat_drop = nn.Dropout(p=dropout_rate)
 
         # extract ntypes that have corresponding metapath
-        # If there're only metapaths like ['MAM', 'MDM'], 'A' and 'D' have no metapath.
-        self.meta_ntypes = set([metapath[0] for metapath in metapath_list])
+        # If there're only metapaths like ['MAM', 'MDM'], 'A' and 'D' have no metapath, so that 'A' and 'D' shouldn't
+        # be considered as nodes that need to aggregate information from metapath.
+        self.dst_ntypes = set([metapath[0] for metapath in metapath_list])
 
         # hidden layers
         self.layers = nn.ModuleList()
         for i in range(num_layers - 1):
             self.layers.append(
                 MAGNN_layer(in_feats=h_feats, inter_attn_feats=inter_attn_feats, out_feats=h_feats, num_heads=num_heads,
                             metapath_list=metapath_list, ntypes=self.ntypes, edge_type_list=edge_type_list,
-                            meta_ntypes=self.meta_ntypes, encoder_type=encoder_type, last_layer=False))
+                            dst_ntypes=self.dst_ntypes, encoder_type=encoder_type, last_layer=False))
 
         # output layer
         self.layers.append(
             MAGNN_layer(in_feats=h_feats, inter_attn_feats=inter_attn_feats, out_feats=num_classes, num_heads=num_heads,
                         metapath_list=metapath_list, ntypes=self.ntypes, edge_type_list=edge_type_list,
-                        meta_ntypes=self.meta_ntypes, encoder_type=encoder_type, last_layer=True))
+                        dst_ntypes=self.dst_ntypes, encoder_type=encoder_type, last_layer=True))
 
-        self.metapath_idx_dict = mp_instances
+        self.metapath_idx_dict = metapath_idx_dict
+
+    def mini_reset_params(self, new_metapth_idx_dict):
+        '''
+
+        Description
+        -----------
+        This method is utilized for reset some parameters including metapath_idx_dict, metapath_list, dst_ntypes...
+        Other Parameters like weight matrix don't need to be updated.
+
+        '''
+        if not self.is_backup: # the params of the original graph has not been stored
+            self.backup['metapath_idx_dict'] = self.metapath_idx_dict
+            self.backup['metapath_list'] = self.metapath_list
+            self.backup['dst_ntypes'] = self.dst_ntypes
+            self.is_backup = True
+
+        self.metapath_idx_dict = new_metapth_idx_dict
+        self.metapath_list = list(new_metapth_idx_dict.keys())
+        self.dst_ntypes = set([meta[0] for meta in self.metapath_list])
+
+        for layer in self.layers:
+            layer.metapath_list = self.metapath_list
+            layer.dst_ntypes = self.dst_ntypes
+
+    def restore_params(self):
```

Comment:
Same.

Reply:
> Same.

When we need to train on subgraphs, we use `mini_reset_params` to reset some params of model based on subgraphs. After the training phase, if we need to validate or test model performance on the whole graph, we use `restore_params` to restore the params based on the whole graph. Training on different graphs need different corresponding parameters of the model.

Reply:
Now that you uploaded the training script, I still didn't see the usage of this method.

Reply:
I don't use this method in training script. Because the graph is too large to directly train the model. In the validate and test phase I still employ mini batch. So we don't need to invoke `restore_params` to restore params of model based on the whole graph.

Moreover, we just need to change `model.mini_reset_params(..)` to `model.restore_params()` in validate and test part if we wanna validate or test on the whole graph.